devices/flash-nrf9160/flashdrv.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Phoenix-RTOS
+ *
+ * plo - operating system loader
+ *
+ * NRF9160 Flash driver
+ *
+ * Copyright 2020, 2022 Phoenix Systems
+ * Author: Aleksander Kaminski, Damian Loewnau
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+/* Copy of flash-stm32 code */
+
+#include <hal/hal.h>
+#include <lib/errno.h>
+#include <devices/devs.h>
+
+
+static const struct {
+	u32 start;
+	u32 end;
+} flashParams[1] = {
+	{ FLASH_PROGRAM_1_ADDR, FLASH_PROGRAM_1_ADDR + FLASH_PROGRAM_BANK_SIZE }
+};
+
+
+static int flashdrv_isValidAddress(unsigned int minor, u32 off, size_t size)
+{
+	size_t fsize = flashParams[minor].end - flashParams[minor].start;
+
+	if (off < fsize && (off + size) <= fsize) {
+		return 1;
+	}
+
+	return 0;
+}
+
+
+static int flashdrv_isValidMinor(unsigned int minor)
+{
+	return minor == 0;
```

Comment:
MISRA: bool -> int cast
```suggestion
	return (minor == 0) ? 1 : 0;
```